federation/manifests/federation-controller-manager-deployment.yaml

```diff
@@ -17,16 +17,23 @@ spec:
       - name: ssl-certs
         hostPath:
           path: /etc/ssl/certs
+      - name: kubeconfig
+        secret:
+          secretName: federation-apiserver-kubeconfig
```

Comment:
If the secret doesn't exist the pod fails to come up. That's not the problem we are trying to solve with the deprecated secret name. We have the deprecated secret name to give an upgrade path to the existing users, so that federation still works if they just update their existing deployment manifest with the new image tag.

They are using this manifest if they are deploying federation from scratch and in that case they are expected to have this secret. It is like a pre-requisite for the new deployments.

I made one minor change though, please take a look.